No security specified. Not doing anything

I want to change the default password for admin user. So I create two secrets below.
`kubectl create secret generic admin-secret --from-literal=username=xxx --from-literal=password=xxx -n opensearch-system`
`kubectl create secret generic securityconfig-secret --from-file=config.yml=config.yml -n opensearch-system`

config.yml : 
```
_meta:
  type: "config"
  config_version: 2

config:
  dynamic:
    # Set filtered_alias_mode to 'disallow' to forbid more than 2 filtered aliases per index
    # Set filtered_alias_mode to 'warn' to allow more than 2 filtered aliases per index but warns about it (default)
    # Set filtered_alias_mode to 'nowarn' to allow more than 2 filtered aliases per index silently
    #filtered_alias_mode: warn
    #do_not_fail_on_forbidden: false
    kibana:
    # Kibana multitenancy
      multitenancy_enabled: true
      server_username: kibanaserver
    #index: '.kibana'
    http:
      anonymous_auth_enabled: false
      xff:
        enabled: false
        internalProxies: '192\.168\.0\.10|192\.168\.0\.11' # regex pattern
        #internalProxies: '.*' # trust all internal proxies, regex pattern
        #remoteIpHeader:  'x-forwarded-for'
        ###### see https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html for regex help
        ###### more information about XFF https://en.wikipedia.org/wiki/X-Forwarded-For
        ###### and here https://tools.ietf.org/html/rfc7239
        ###### and https://tomcat.apache.org/tomcat-8.0-doc/config/valve.html#Remote_IP_Valve
    authc:
      kerberos_auth_domain:
        http_enabled: false
        transport_enabled: false
        order: 6
        http_authenticator:
          type: kerberos
          challenge: true
          config:
            # If true a lot of kerberos/security related debugging output will be logged to standard out
            krb_debug: false
            # If true then the realm will be stripped from the user name
            strip_realm_from_principal: true
        authentication_backend:
          type: noop
      basic_internal_auth_domain:
        description: "Authenticate via HTTP Basic against internal users database"
        http_enabled: true
        transport_enabled: true
        order: 4
        http_authenticator:
          type: basic
          challenge: true
        authentication_backend:
          type: intern
      proxy_auth_domain:
        description: "Authenticate via proxy"
        http_enabled: false
        transport_enabled: false
        order: 3
        http_authenticator:
          type: proxy
          challenge: false
          config:
            user_header: "x-proxy-user"
            roles_header: "x-proxy-roles"
        authentication_backend:
          type: noop
      jwt_auth_domain:
        description: "Authenticate via Json Web Token"
        http_enabled: false
        transport_enabled: false
        order: 0
        http_authenticator:
          type: jwt
          challenge: false
          config:
            signing_key: "base64 encoded HMAC key or public RSA/ECDSA pem key"
            jwt_header: "Authorization"
            jwt_url_parameter: null
            roles_key: null
            subject_key: null
        authentication_backend:
          type: noop
      clientcert_auth_domain:
        description: "Authenticate via SSL client certificates"
        http_enabled: false
        transport_enabled: false
        order: 2
        http_authenticator:
          type: clientcert
          config:
            username_attribute: cn #optional, if omitted DN becomes username
          challenge: false
        authentication_backend:
          type: noop
      ldap:
        description: "Authenticate via LDAP or Active Directory"
        http_enabled: false
        transport_enabled: false
        order: 5
        http_authenticator:
          type: basic
          challenge: false
        authentication_backend:
          # LDAP authentication backend (authenticate users against a LDAP or Active Directory)
          type: ldap
          config:
            # enable ldaps
            enable_ssl: false
            # enable start tls, enable_ssl should be false
            enable_start_tls: false
            # send client certificate
            enable_ssl_client_auth: false
            # verify ldap hostname
            verify_hostnames: true
            hosts:
            - localhost:8389
            bind_dn: null
            password: null
            userbase: 'ou=people,dc=example,dc=com'
            # Filter to search for users (currently in the whole subtree beneath userbase)
            # {0} is substituted with the username
            usersearch: '(sAMAccountName={0})'
            # Use this attribute from the user as username (if not set then DN is used)
            username_attribute: null
    authz:
      roles_from_myldap:
        description: "Authorize via LDAP or Active Directory"
        http_enabled: false
        transport_enabled: false
        authorization_backend:
          # LDAP authorization backend (gather roles from a LDAP or Active Directory, you have to configure the above LDAP authentication backend settings too)
          type: ldap
          config:
            # enable ldaps
            enable_ssl: false
            # enable start tls, enable_ssl should be false
            enable_start_tls: false
            # send client certificate
            enable_ssl_client_auth: false
            # verify ldap hostname
            verify_hostnames: true
            hosts:
            - localhost:8389
            bind_dn: null
            password: null
            rolebase: 'ou=groups,dc=example,dc=com'
            # Filter to search for roles (currently in the whole subtree beneath rolebase)
            # {0} is substituted with the DN of the user
            # {1} is substituted with the username
            # {2} is substituted with an attribute value from user's directory entry, of the authenticated user. Use userroleattribute to specify the name of the attribute
            rolesearch: '(member={0})'
            # Specify the name of the attribute which value should be substituted with {2} above
            userroleattribute: null
            # Roles as an attribute of the user entry
            userrolename: disabled
            #userrolename: memberOf
            # The attribute in a role entry containing the name of that role, Default is "name".
            # Can also be "dn" to use the full DN as rolename.
            rolename: cn
            # Resolve nested roles transitive (roles which are members of other roles and so on ...)
            resolve_nested_roles: true
            userbase: 'ou=people,dc=example,dc=com'
            # Filter to search for users (currently in the whole subtree beneath userbase)
            # {0} is substituted with the username
            usersearch: '(uid={0})'
            # Skip users matching a user name, a wildcard or a regex pattern
            #skip_users:
            #  - 'cn=Michael Jackson,ou*people,o=TEST'
            #  - '/\S*/'
      roles_from_another_ldap:
        description: "Authorize via another Active Directory"
        http_enabled: false
        transport_enabled: false
        authorization_backend:
          type: ldap
          #config goes here ...
  #    auth_failure_listeners:
  #      ip_rate_limiting:
  #        type: ip
  #        allowed_tries: 10
  #        time_window_seconds: 3600
  #        block_expiry_seconds: 600
  #        max_blocked_clients: 100000
  #        max_tracked_clients: 100000
  #      internal_authentication_backend_limiting:
  #        type: username
  #        authentication_backend: intern
  #        allowed_tries: 10
  #        time_window_seconds: 3600
  #        block_expiry_seconds: 600
  #        max_blocked_clients: 100000
  #        max_tracked_clients: 100000
```

my cluster,yaml:
```
[root@eks-test opensearch]# cat cluster.yaml
apiVersion: opensearch.opster.io/v1
kind: OpenSearchCluster
metadata:
  name: opensearch
  namespace: opensearch-system
spec:
  general:
    serviceName: opensearch
    version: 1.3.1
  security:
    config:  # Everything related to the securityconfig
      securityConfigSecret:
        name:  securityconfig-secret
      #adminSecret:
      #  name:  # Name of a secret that contains the admin client certificate
      adminCredentialsSecret:
        name:  admin-secret
  dashboards:
    enable: true
    version: 1.3.1
    replicas: 1
    #additionalConfig:
    #  opensearch_security.multitenancy.enabled: "true"
    resources:
      requests:
         memory: "512Mi"
         cpu: "200m"
      limits:
         memory: "512Mi"
         cpu: "200m"
  nodePools:
    - component: masters-data-hot
      replicas: 2
      diskSize: "5Gi"
      NodeSelector:
      resources:
         requests:
            memory: "1Gi"
            cpu: "500m"
         limits:
            memory: "1Gi"
            cpu: "500m"
      roles:
        - "data"
        - "master"
        - "ingest"
      additionalConfig:
        node.attr.type: hot
    - component: masters-data-cold
      replicas: 1
      diskSize: "5Gi"
      NodeSelector:
      resources:
         requests:
            memory: "1Gi"
            cpu: "500m"
         limits:
            memory: "1Gi"
            cpu: "500m"
      roles:
        - "data"
        - "master"
        - "ingest"
      additionalConfig:
        node.attr.type: cold
```

After all I got error log in opensearch-operator-controller-manager:

```
1.6540534730476532e+09  INFO    controller.opensearchcluster    Starting EventSource    {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster", "source": "kind source: *v1.OpenSearchCluster"}
1.6540534730476897e+09  INFO    controller.opensearchcluster    Starting EventSource    {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster", "source": "kind source: *v1.Pod"}
1.6540534730476956e+09  INFO    controller.opensearchcluster    Starting EventSource    {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster", "source": "kind source: *v1.Secret"}
1.654053473047701e+09   INFO    controller.opensearchcluster    Starting EventSource    {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster", "source": "kind source: *v1.ConfigMap"}
1.6540534730477064e+09  INFO    controller.opensearchcluster    Starting EventSource    {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster", "source": "kind source: *v1.Service"}
1.654053473047712e+09   INFO    controller.opensearchcluster    Starting EventSource    {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster", "source": "kind source: *v1.Deployment"}
1.6540534730471847e+09  DEBUG   events  Normal  {"object": {"kind":"ConfigMap","namespace":"opensearch-system","name":"a867c7dc.opensearch.opster.io","uid":"091c31f1-6cfe-434c-a42b-addd75d3cf93","apiVersion":"v1","resourceVersion":"21425898"}, "reason": "LeaderElection", "message": "opensearch-operator-controller-manager-597cbf7fd4-ct8w9_819e05ef-012a-4dc2-9da0-c94c5383a72b became leader"}
1.6540534730477169e+09  INFO    controller.opensearchcluster    Starting EventSource    {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster", "source": "kind source: *v1.StatefulSet"}
1.6540534730477664e+09  INFO    controller.opensearchcluster    Starting Controller     {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster"}
1.6540534730477257e+09  DEBUG   events  Normal  {"object": {"kind":"Lease","namespace":"opensearch-system","name":"a867c7dc.opensearch.opster.io","uid":"883dc988-8d05-4551-baaf-4daa6fa495dd","apiVersion":"coordination.k8s.io/v1","resourceVersion":"21425900"}, "reason": "LeaderElection", "message": "opensearch-operator-controller-manager-597cbf7fd4-ct8w9_819e05ef-012a-4dc2-9da0-c94c5383a72b became leader"}
1.654053473149658e+09   INFO    controller.opensearchcluster    Starting workers        {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster", "worker count": 1}
1.6540534731497726e+09  INFO    controller.opensearchcluster    Reconciling OpenSearchCluster   {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster", "name": "opensearch", "namespace": "opensearch-system", "cluster": "opensearch-system/opensearch"}
1.654053473162388e+09   INFO    controller.opensearchcluster    No security specified. Not doing anything       {"reconciler group": "opensearch.opster.io", "reconciler kind": "OpenSearchCluster", "name": "opensearch", "namespace": "opensearch-system"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1548d29]

goroutine 397 [running]:
opensearch.opster.io/pkg/reconcilers.(*SecurityconfigReconciler).determineAdminSecret(0x2)
        /workspace/pkg/reconcilers/securityconfig.go:196 +0x49
opensearch.opster.io/pkg/reconcilers.(*SecurityconfigReconciler).Reconcile(0xc000511180)
        /workspace/pkg/reconcilers/securityconfig.go:67 +0x48
opensearch.opster.io/controllers.(*OpenSearchClusterReconciler).reconcilePhaseRunning(0xc0000880a0, {0x1b64018, 0xc000bda870})
        /workspace/controllers/opensearchController.go:326 +0x6ae
opensearch.opster.io/controllers.(*OpenSearchClusterReconciler).Reconcile(0xc0000880a0, {0x1b64018, 0xc000bda870}, {{{0xc0004dcae0, 0x11}, {0xc0004d82e0, 0xa}}})
        /workspace/controllers/opensearchController.go:141 +0x4be
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc00015c2c0, {0x1b64018, 0xc000bda7b0}, {{{0xc0004dcae0, 0x184b580}, {0xc0004d82e0, 0x413a94}}})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114 +0x26f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00015c2c0, {0x1b63f70, 0xc0005da400}, {0x178ff00, 0xc0003b8020})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311 +0x33e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00015c2c0, {0x1b63f70, 0xc0005da400})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:223 +0x357
```

--- a/opensearch-operator/pkg/reconcilers/securityconfig.go
+++ b/opensearch-operator/pkg/reconcilers/securityconfig.go
@@ -193,7 +193,7 @@ func checksum(data map[string][]byte) (string, error) {
 func (r *SecurityconfigReconciler) determineAdminSecret() string {
 	if r.instance.Spec.Security.Config != nil && r.instance.Spec.Security.Config.AdminSecret.Name != "" {
 		return r.instance.Spec.Security.Config.AdminSecret.Name
-	} else if r.instance.Spec.Security.Tls.Transport.Generate {
+	} else if r.instance.Spec.Security.Tls != nil && r.instance.Spec.Security.Tls.Transport != nil && r.instance.Spec.Security.Tls.Transport.Generate {
 		return fmt.Sprintf("%s-admin-cert", r.instance.Name)
 	} else {
 		return ""

--- a/opensearch-operator/pkg/reconcilers/securityconfig_test.go
+++ b/opensearch-operator/pkg/reconcilers/securityconfig_test.go
@@ -76,6 +76,7 @@ var _ = Describe("Securityconfig Reconciler", func() {
 
 	Context("When Reconciling the securityconfig reconciler with securityconfig secret configured and available", func() {
 		It("should start an update job", func() {
+			var clusterName = "securityconfig"
 			// Create namespace and secrets first
 			Expect(CreateNamespace(k8sClient, clusterName)).Should(Succeed())
 			configSecret := corev1.Secret{
@@ -119,6 +120,46 @@ var _ = Describe("Securityconfig Reconciler", func() {
 				err := k8sClient.Get(context.Background(), client.ObjectKey{Name: clusterName + "-securityconfig-update", Namespace: clusterName}, &job)
 				return err == nil
 			}, timeout, interval).Should(BeTrue())
+
+		})
+	})
+
+	Context("When Reconciling the securityconfig reconciler with securityconfig secret but no adminSecret configured", func() {
+		It("should not start an update job", func() {
+			var clusterName = "securityconfig-noadminsecret"
+			// Create namespace and secret first
+			Expect(CreateNamespace(k8sClient, clusterName)).Should(Succeed())
+			configSecret := corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: "securityconfig", Namespace: clusterName},
+				StringData: map[string]string{"config.yml": "foobar"},
+			}
+			err := k8sClient.Create(context.Background(), &configSecret)
+			Expect(err).ToNot(HaveOccurred())
+
+			spec := opsterv1.OpenSearchCluster{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName, Namespace: clusterName, UID: "dummyuid"},
+				Spec: opsterv1.ClusterSpec{
+					General: opsterv1.GeneralConfig{},
+					Security: &opsterv1.Security{
+						Config: &opsterv1.SecurityConfig{
+							SecurityconfigSecret: corev1.LocalObjectReference{Name: "securityconfig"},
+						},
+					},
+				}}
+
+			reconcilerContext := NewReconcilerContext(spec.Spec.NodePools)
+			underTest := NewSecurityconfigReconciler(
+				k8sClient,
+				context.Background(),
+				&helpers.MockEventRecorder{},
+				&reconcilerContext,
+				&spec,
+			)
+			_, err = underTest.Reconcile()
+			Expect(err).ToNot(HaveOccurred())
+
+			job := batchv1.Job{}
+			Expect(k8sClient.Get(context.Background(), client.ObjectKey{Name: clusterName + "-securityconfig-update", Namespace: clusterName}, &job)).To(HaveOccurred())
 
 		})
 	})

--- a/opensearch-operator/pkg/reconcilers/tls.go
+++ b/opensearch-operator/pkg/reconcilers/tls.go
@@ -360,8 +360,8 @@ func (r *TLSReconciler) handleHttp() error {
 
 		var ca tls.Cert
 		var err error
-		if r.instance.Spec.Security.Tls.Http.TlsCertificateConfig.CaSecret.Name != "" {
-			ca, err = r.providedCaCert(r.instance.Spec.Security.Tls.Http.TlsCertificateConfig.CaSecret.Name, namespace)
+		if tlsConfig.TlsCertificateConfig.CaSecret.Name != "" {
+			ca, err = r.providedCaCert(tlsConfig.TlsCertificateConfig.CaSecret.Name, namespace)
 		} else {
 			ca, err = helpers.ReadOrGenerateCaCert(r.pki, r.Client, r.ctx, r.instance)
 		}

